tests/serverpod_test_server/test_integration/logging/session_logging_test.dart

```diff
@@ -0,0 +1,318 @@
+            consoleEnabled: true,
+          ),
+        ),
+      );
+      await server.start();
+      session = await server.createSession(enableLogging: false);
+      await LoggingUtil.clearAllLogs(session);
+    });
+
+    tearDown(() async {
+      await session.close();
+      await server.shutdown(exitProcess: false);
+    });
+
+    test(
+        'logs should not be stored in the database but should appear in stdout',
+        () async {
+      var settings = RuntimeSettingsBuilder().build();
+      await server.updateRuntimeSettings(settings);
+
+      await IOOverrides.runZoned(
+        () async {
+          var testSession = await server.createSession(enableLogging: true);
+          testSession.log('Test message for console');
+          await testSession.close();
+        },
+        stdout: () => record,
+      );
+
+      var logs = await LoggingUtil.findAllLogs(session);
+      expect(logs, isEmpty);
+      expect(record.output.contains('Test message for console'), isTrue);
+    });
+  });
+
+  group('Given both persistent and console logging are disabled', () {
+    late Serverpod server;
+    late Session session;
+    late MockStdout record;
+
+    setUp(() async {
+      record = MockStdout();
+
+      server = IntegrationTestServer.create(
+        config: ServerpodConfig(
+          runMode: runMode,
+          apiServer: ServerConfig(
+            port: 8080,
+            publicHost: 'localhost',
+            publicPort: 8080,
+            publicScheme: 'http',
+          ),
+          database: DatabaseConfig(
+            host: 'postgres',
+            port: 5432,
+            user: 'postgres',
+            password: 'password',
+            name: 'serverpod_test',
+          ),
+          sessionLogs: SessionLogConfig(
+            persistentEnabled: false,
+            consoleEnabled: false,
+          ),
+        ),
+      );
+      await server.start();
+      session = await server.createSession(enableLogging: false);
+      await LoggingUtil.clearAllLogs(session);
+    });
+
+    tearDown(() async {
+      await session.close();
+      await server.shutdown(exitProcess: false);
+    });
+
+    test('no logs should be written to the database or stdout', () async {
+      var settings = RuntimeSettingsBuilder().build();
+      await server.updateRuntimeSettings(settings);
+
+      await IOOverrides.runZoned(
+        () async {
+          var testSession = await server.createSession(enableLogging: true);
+          testSession.log('Test message');
+          await testSession.close();
+        },
+        stdout: () => record,
+      );
+
+      var logs = await LoggingUtil.findAllLogs(session);
+      expect(logs, isEmpty);
+      expect(record.output.contains('Test message'), isFalse);
+    });
+  });
+
+  group(
+      'Given persistent logging is enabled but the database support is not available',
+      () {
+    test('a StateError should be thrown due to lack of database support',
```

Comment:
Suggestion: I think these two can be merged into a single `test`.

Reply:
@hampuslavin done:

https://github.com/serverpod/serverpod/blob/cea16cfc2c0a9b38c93dfed2b27836a04c24bd53/tests/serverpod_test_server/test_integration/logging/session_logging_test.dart#L215-L244